flux-attrs/src/lib.rs

```diff
@@ -1,21 +1,240 @@
+                ))
+            }
+        }
+    }).collect::<syn::Result<Vec<syn::ImplItem>>>()?;
+    Ok(TokenStream::from_iter(
+        [item_struct.to_token_stream(), dummy_impl.to_token_stream()].into_iter(),
+    ))
+}
+
+fn create_dummy_ident(dummy_prefix: &mut String, ty: &syn::Type) -> syn::Result<syn::Ident> {
+    use syn::Type::*;
+    match ty {
+        Reference(ty_ref) => {
+            if ty_ref.mutability.is_some() {
+                dummy_prefix.push_str("Mut");
+            };
+            dummy_prefix.push_str("Ref");
+            create_dummy_ident(dummy_prefix, ty_ref.elem.as_ref())
+        }
+        Slice(ty_slice) => {
+            dummy_prefix.push_str("Slice");
+            create_dummy_ident(dummy_prefix, ty_slice.elem.as_ref())
+        }
+        // For paths, we mangle the last identifier
+        Path(ty_path) => {
+            if let Some(path_segment) = ty_path.path.segments.last() {
+                // Mangle the identifier using the dummy_prefix
+                let ident = syn::Ident::new(
+                    &format!("{}{}", dummy_prefix, path_segment.ident),
+                    path_segment.ident.span(),
+                );
+                Ok(ident)
+            } else {
+                Err(syn::Error::new(
+                    ty_path.path.span(),
+                    format!("Invalid extern_spec: empty TypePath {:?}", ty_path.path),
+                ))
+            }
+        }
+        _ => {
+            Err(syn::Error::new(
+                ty.span(),
+                format!("Invalid extern_spec: unsupported type {:?}", ty),
+            ))
+        }
+    }
+}
+
+/// Create a dummy struct with a single unnamed field that is the external struct.
+///
+/// Example:
+///
+/// ```ignore
+/// #[extern_spec(std::vec)]
+/// #[flux::refined_by(n: int)]
+/// struct Vec<T>;
+///
+/// =>
+///
+/// #[flux::extern_spec]
+/// #[allow(unused, dead_code)]
+/// #[flux::refined_by(n: int)]
+/// struct FluxExternStructVec<T>(std::vec::Vec<T>);
+/// ```
+fn create_dummy_struct(
+    mod_path: Option<syn::Path>,
+    item_struct: syn::ItemStruct,
+) -> syn::Result<TokenStream> {
+    let item_struct_span = item_struct.span();
+    let fields_span = item_struct.fields.span();
+    if let syn::Fields::Unit = item_struct.fields {
+        let mut dummy_struct = item_struct.clone();
+        let ident = item_struct.ident;
+        let generics = item_struct.generics;
+        dummy_struct.ident = format_ident!("FluxExternStruct{}", ident);
+        dummy_struct.semi_token = None;
+        let dummy_field: syn::FieldsUnnamed = if let Some(mod_path) = mod_path {
+            parse_quote_spanned! {item_struct_span =>
+                ( #mod_path :: #ident #generics )
+            }
+        } else {
+            parse_quote_spanned! {item_struct_span =>
+                ( #ident #generics )
+            }
+        };
```

Comment:
I want to be able to just put `#mod_path`, but it seems to error for a type like `::Vec<T>`.

There are other cases where I have to special-case this logic that boil down the the same issue, e.g.

```rust
::&[T]
```

is not a valid type but

```rust
&[T]
```

is.